RecoTracker/MkFit/plugins/MkFitProducer.cc

```diff
@@ -0,0 +1,148 @@
+  bool backwardFitInCMSSW_;
+  bool mkfitSilent_;
+};
+
+MkFitProducer::MkFitProducer(edm::ParameterSet const& iConfig)
+    : hitsSeedsToken_{consumes<MkFitInputWrapper>(iConfig.getParameter<edm::InputTag>("hitsSeeds"))},
+      geomToken_{esConsumes<TrackerGeometry, TrackerDigiGeometryRecord>()},
+      putToken_{produces<MkFitOutputWrapper>()},
+      backwardFitInCMSSW_{iConfig.getParameter<bool>("backwardFitInCMSSW")},
+      mkfitSilent_{iConfig.getUntrackedParameter<bool>("mkfitSilent")} {
+  const auto build = iConfig.getParameter<std::string>("buildingRoutine");
+  bool isFV = false;
+  if (build == "bestHit") {
+    buildFunction_ = mkfit::runBuildingTestPlexBestHit;
+  } else if (build == "standard") {
+    buildFunction_ = mkfit::runBuildingTestPlexStandard;
+  } else if (build == "cloneEngine") {
+    buildFunction_ = mkfit::runBuildingTestPlexCloneEngine;
+  } else if (build == "fullVector") {
+    isFV = true;
+    buildFunction_ = mkfit::runBuildingTestPlexFV;
+  } else {
+    throw cms::Exception("Configuration") << "Invalid value for parameter 'buildingRoutine' " << build
+                                          << ", allowed are bestHit, standard, cloneEngine, fullVector";
+  }
+
+  const auto seedClean = iConfig.getParameter<std::string>("seedCleaning");
+  auto seedCleanOpt = mkfit::ConfigWrapper::SeedCleaningOpts::noCleaning;
+  if (seedClean == "none") {
+    seedCleanOpt = mkfit::ConfigWrapper::SeedCleaningOpts::noCleaning;
+  } else if (seedClean == "N2") {
+    seedCleanOpt = mkfit::ConfigWrapper::SeedCleaningOpts::cleanSeedsN2;
+  } else {
+    throw cms::Exception("Configuration")
+        << "Invalida value for parameter 'seedCleaning' " << seedClean << ", allowed are none, N2";
+  }
+
+  auto backwardFitOpt =
+      backwardFitInCMSSW_ ? mkfit::ConfigWrapper::BackwardFit::noFit : mkfit::ConfigWrapper::BackwardFit::toFirstLayer;
+
+  // TODO: what to do when we have multiple instances of MkFitProducer in a job?
+  mkfit::MkBuilderWrapper::populate(isFV);
+  mkfit::ConfigWrapper::initializeForCMSSW(seedCleanOpt, backwardFitOpt, mkfitSilent_);
+}
+
+void MkFitProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+
+  desc.add("hitsSeeds", edm::InputTag("mkFitInputConverter"));
+  desc.add<std::string>("buildingRoutine", "cloneEngine")
+      ->setComment("Valid values are: 'bestHit', 'standard', 'cloneEngine', 'fullVector'");
+  desc.add<std::string>("seedCleaning", "N2")->setComment("Valid values are: 'none', 'N2'");
+  desc.add("backwardFitInCMSSW", false)
+      ->setComment("Do backward fit (to innermost hit) in CMSSW (true) or mkFit (false)");
+  desc.addUntracked("mkfitSilent", true)->setComment("Allows to enables printouts from mkfit with 'False'");
+
+  descriptions.add("mkFitProducer", desc);
+}
+
+std::unique_ptr<mkfit::MkBuilderWrapper> MkFitProducer::beginStream(edm::StreamID iID) const {
+  return std::make_unique<mkfit::MkBuilderWrapper>();
```

Comment:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-e6d66d/2241/llvm-analysis/report-d8d5d6.html#EndPath
static analyzer is not happy
``` 4  ←  Potential leak of memory pointed to by field '_M_head_impl'``` 
false positive?